hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/customfees/AssessmentResult.java

```diff
@@ -30,37 +30,43 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 import org.apache.commons.lang3.tuple.Pair;
 
 public class AssessmentResult {
     private Map<TokenID, Map<AccountID, Long>> htsAdjustments;
     // two maps to aggregate all custom fee balance changes. These two maps are used
     // to construct a transaction body that needs to be assessed again for custom fees
     private Map<AccountID, Long> hbarAdjustments;
-    // Any debits in this set should not trigger custom fee charging again
-    private Set<TokenID> exemptDebits;
     private Set<Pair<AccountID, TokenID>> royaltiesPaid;
-    private Map<TokenID, Map<AccountID, Long>> inputTokenAdjustments;
-
+    private Map<TokenID, Map<AccountID, Long>> immutableInputTokenAdjustments;
+    private Map<TokenID, Map<AccountID, Long>> mutableInputTokenAdjustments;
     private Map<AccountID, Long> inputHbarAdjustments;
     /* And for each "assessable change" that can be charged a custom fee, delegate to our
     fee assessor to update the balance changes with the custom fee. */
     private List<AssessedCustomFee> assessedCustomFees;
 
     public AssessmentResult(
             final List<TokenTransferList> inputTokenTransfers, final List<AccountAmount> inputHbarTransfers) {
-        inputTokenAdjustments = buildFungibleTokenTransferMap(inputTokenTransfers);
+        mutableInputTokenAdjustments = buildFungibleTokenTransferMap(inputTokenTransfers);
+        immutableInputTokenAdjustments = mutableInputTokenAdjustments.entrySet().stream()
+                .collect(Collectors.collectingAndThen(
```

Comment:
_Nit_ - the `Collector` factories `collectingAndThen()`, `reducing()`, `filtering()`, etc. are only meant to be used as "downstream" collectors in for a [`Collectors.groupingBy()`](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#groupingBy-java.util.function.Function-java.util.stream.Collector-) usage.

Within a normal stream, you should be able to perform the desired aggregation on the source stream, e.g.
```
immutableInputTokenAdjustments = Collections.unmodifiableMap(
    mutableInputTokenAdjustments.entrySet().stream()
        .collect(Collectors.toMap(Map.Entry::getKey, entry -> Map.copyOf(entry.getValue()))));
```

Reply:
yes, will make that change Michael. tyvm !